### ISSUE TITLE
things not exported by HoTT.v

### ISSUE THREAD

**Issue by jdchristensen** [HoTT/HoTT#1511]
As a test, I did:
```
make clean
make theories/HoTT.vo
make
```
to see what things in the library were not used by HoTT.vo.  It turns out that there are many things.  I know that some are intentional, but I wonder if some of these have been accidentally left out?  The list is below, sorted alphabetically.  (BTW, building HoTT.vo only takes 54% of the total build time.)

<details><summary>Files built after HoTT.vo</summary>

```
contrib/HoTTBookExercises.v
contrib/HoTTBook.v
theories/Algebra/Groups/FreeGroup/LoopSusp.v
theories/Algebra/Groups/FreeProduct.v
theories/Algebra/Groups/GroupCoeq.v
theories/Algebra/Groups/Presentation.v
theories/Algebra/Universal/Algebra.v
theories/Algebra/Universal/Congruence.v
theories/Algebra/Universal/Homomorphism.v
theories/Algebra/Universal/Operation.v
theories/Algebra/Universal/TermAlgebra.v
theories/Analysis/Locator.v
theories/Categories/Adjoint/Composition/AssociativityLaw.v
theories/Categories/Adjoint/Composition/Core.v
theories/Categories/Adjoint/Composition/IdentityLaws.v
theories/Categories/Adjoint/Composition/LawsTactic.v
theories/Categories/Adjoint/Composition.v
theories/Categories/Adjoint/Core.v
theories/Categories/Adjoint/Dual.v
theories/Categories/Adjoint/Functorial/Core.v
theories/Categories/Adjoint/Functorial/Laws.v
theories/Categories/Adjoint/Functorial/Parts.v
theories/Categories/Adjoint/Functorial.v
theories/Categories/Adjoint/HomCoercions.v
theories/Categories/Adjoint/Hom.v
theories/Categories/Adjoint/Identity.v
theories/Categories/Adjoint/Notations.v
theories/Categories/Adjoint/Paths.v
theories/Categories/Adjoint/Pointwise.v
theories/Categories/Adjoint/UnitCounitCoercions.v
theories/Categories/Adjoint/UnitCounit.v
theories/Categories/Adjoint/UniversalMorphisms/Core.v
theories/Categories/Adjoint/UniversalMorphisms.v
theories/Categories/Adjoint/Utf8.v
theories/Categories/Adjoint.v
theories/Categories/Cat/Core.v
theories/Categories/Category/Core.v
theories/Categories/Category/Dual.v
theories/Categories/Category/Morphisms.v
theories/Categories/Category/Notations.v
theories/Categories/Category/Objects.v
theories/Categories/CategoryOfGroupoids.v
theories/Categories/CategoryOfSections/Core.v
theories/Categories/CategoryOfSections.v
theories/Categories/Category/Paths.v
theories/Categories/Category/Pi.v
theories/Categories/Category/Prod.v
theories/Categories/Category/Sigma/Core.v
theories/Categories/Category/Sigma/OnMorphisms.v
theories/Categories/Category/Sigma/OnObjects.v
theories/Categories/Category/Sigma/Univalent.v
theories/Categories/Category/Sigma.v
theories/Categories/Category/Strict.v
theories/Categories/Category/Subcategory/Full.v
theories/Categories/Category/Subcategory.v
theories/Categories/Category/Subcategory/Wide.v
theories/Categories/Category/Sum.v
theories/Categories/Category/Univalent.v
theories/Categories/Category/Utf8.v
theories/Categories/Category.v
theories/Categories/Cat/Morphisms.v
theories/Categories/Cat.v
theories/Categories/ChainCategory.v
theories/Categories/Comma/Core.v
theories/Categories/Comma/Dual.v
theories/Categories/Comma/Functorial.v
theories/Categories/Comma/InducedFunctors.v
theories/Categories/Comma/Notations.v
theories/Categories/Comma/ProjectionFunctors.v
theories/Categories/Comma/Projection.v
theories/Categories/Comma/Utf8.v
theories/Categories/Comma.v
theories/Categories/DependentProduct.v
theories/Categories/DiscreteCategory.v
theories/Categories/DualFunctor.v
theories/Categories/ExponentialLaws/Law0.v
theories/Categories/ExponentialLaws/Law1/Functors.v
theories/Categories/ExponentialLaws/Law1/Law.v
theories/Categories/ExponentialLaws/Law1.v
theories/Categories/ExponentialLaws/Law2/Functors.v
theories/Categories/ExponentialLaws/Law2/Law.v
theories/Categories/ExponentialLaws/Law2.v
theories/Categories/ExponentialLaws/Law3/Functors.v
theories/Categories/ExponentialLaws/Law3/Law.v
theories/Categories/ExponentialLaws/Law3.v
theories/Categories/ExponentialLaws/Law4/Functors.v
theories/Categories/ExponentialLaws/Law4/Law.v
theories/Categories/ExponentialLaws/Law4.v
theories/Categories/ExponentialLaws/Tactics.v
theories/Categories/ExponentialLaws.v
theories/Categories/Functor/Attributes.v
theories/Categories/FunctorCategory/Core.v
theories/Categories/FunctorCategory/Dual.v
theories/Categories/FunctorCategory/Functorial.v
theories/Categories/FunctorCategory/Morphisms.v
theories/Categories/FunctorCategory/Notations.v
theories/Categories/FunctorCategory/Utf8.v
theories/Categories/FunctorCategory.v
theories/Categories/Functor/Composition/Core.v
theories/Categories/Functor/Composition/Functorial/Attributes.v
theories/Categories/Functor/Composition/Functorial/Core.v
theories/Categories/Functor/Composition/Functorial.v
theories/Categories/Functor/Composition/Laws.v
theories/Categories/Functor/Composition.v
theories/Categories/Functor/Core.v
theories/Categories/Functor/Dual.v
theories/Categories/Functor/Identity.v
theories/Categories/Functor/Notations.v
theories/Categories/Functor/Paths.v
theories/Categories/Functor/Pointwise/Core.v
theories/Categories/Functor/Pointwise/Properties.v
theories/Categories/Functor/Pointwise.v
theories/Categories/Functor/Prod/Core.v
theories/Categories/Functor/Prod/Functorial.v
theories/Categories/Functor/Prod/Universal.v
theories/Categories/Functor/Prod.v
theories/Categories/Functor/Sum.v
theories/Categories/Functor/Utf8.v
theories/Categories/Functor.v
theories/Categories/FundamentalPreGroupoidCategory.v
theories/Categories/Grothendieck/PseudofunctorToCat.v
theories/Categories/Grothendieck/ToCat.v
theories/Categories/Grothendieck/ToSet/Core.v
theories/Categories/Grothendieck/ToSet/Morphisms.v
theories/Categories/Grothendieck/ToSet/Univalent.v
theories/Categories/Grothendieck/ToSet.v
theories/Categories/Grothendieck.v
theories/Categories/GroupoidCategory/Core.v
theories/Categories/GroupoidCategory/Dual.v
theories/Categories/GroupoidCategory/Morphisms.v
theories/Categories/GroupoidCategory.v
theories/Categories/HomFunctor.v
theories/Categories/HomotopyPreCategory.v
theories/Categories/IndiscreteCategory.v
theories/Categories/InitialTerminalCategory/Core.v
theories/Categories/InitialTerminalCategory/Functors.v
theories/Categories/InitialTerminalCategory/NaturalTransformations.v
theories/Categories/InitialTerminalCategory/Notations.v
theories/Categories/InitialTerminalCategory/Pseudofunctors.v
theories/Categories/InitialTerminalCategory.v
theories/Categories/KanExtensions/Core.v
theories/Categories/KanExtensions/Functors.v
theories/Categories/KanExtensions.v
theories/Categories/LaxComma/CoreLaws.v
theories/Categories/LaxComma/CoreParts.v
theories/Categories/LaxComma/Core.v
theories/Categories/LaxComma/Notations.v
theories/Categories/LaxComma/Utf8.v
theories/Categories/LaxComma.v
theories/Categories/Limits/Core.v
theories/Categories/Limits/Functors.v
theories/Categories/Limits.v
theories/Categories/NatCategory.v
theories/Categories/NaturalTransformation/Composition/Core.v
theories/Categories/NaturalTransformation/Composition/Functorial.v
theories/Categories/NaturalTransformation/Composition/Laws.v
theories/Categories/NaturalTransformation/Composition.v
theories/Categories/NaturalTransformation/Core.v
theories/Categories/NaturalTransformation/Dual.v
theories/Categories/NaturalTransformation/Identity.v
theories/Categories/NaturalTransformation/Isomorphisms.v
theories/Categories/NaturalTransformation/Notations.v
theories/Categories/NaturalTransformation/Paths.v
theories/Categories/NaturalTransformation/Pointwise.v
theories/Categories/NaturalTransformation/Prod.v
theories/Categories/NaturalTransformation/Sum.v
theories/Categories/NaturalTransformation/Utf8.v
theories/Categories/NaturalTransformation.v
theories/Categories/Notations.v
theories/Categories/ProductLaws.v
theories/Categories/Profunctor/Core.v
theories/Categories/Profunctor/Identity.v
theories/Categories/Profunctor/Notations.v
theories/Categories/Profunctor/Representable.v
theories/Categories/Profunctor/Utf8.v
theories/Categories/Profunctor.v
theories/Categories/Pseudofunctor/Core.v
theories/Categories/Pseudofunctor/FromFunctor.v
theories/Categories/Pseudofunctor/Identity.v
theories/Categories/Pseudofunctor/RewriteLaws.v
theories/Categories/Pseudofunctor.v
theories/Categories/PseudonaturalTransformation/Core.v
theories/Categories/PseudonaturalTransformation.v
theories/Categories/SemiSimplicialSets.v
theories/Categories/SetCategory/Core.v
theories/Categories/SetCategory/Functors/SetProp.v
theories/Categories/SetCategory/Functors.v
theories/Categories/SetCategory/Morphisms.v
theories/Categories/SetCategory.v
theories/Categories/SimplicialSets.v
theories/Categories/Structure/Core.v
theories/Categories/Structure/IdentityPrinciple.v
theories/Categories/Structure/Notations.v
theories/Categories/Structure/Utf8.v
theories/Categories/Structure.v
theories/Categories/UniversalProperties.v
theories/Categories/Utf8.v
theories/Categories.v
theories/Categories/Yoneda.v
theories/Classes/categories/ua_category.v
theories/Classes/implementations/assume_rationals.v
theories/Classes/implementations/binary_naturals.v
theories/Classes/implementations/bool.v
theories/Classes/implementations/family_prod.v
theories/Classes/implementations/field_of_fractions.v
theories/Classes/implementations/hprop_lattice.v
theories/Classes/implementations/list.v
theories/Classes/implementations/natpair_integers.v
theories/Classes/implementations/ne_list.v
theories/Classes/implementations/peano_naturals.v
theories/Classes/implementations/pointwise.v
theories/Classes/interfaces/archimedean.v
theories/Classes/interfaces/cauchy.v
theories/Classes/interfaces/integers.v
theories/Classes/interfaces/monad.v
theories/Classes/interfaces/naturals.v
theories/Classes/interfaces/orders.v
theories/Classes/interfaces/rationals.v
theories/Classes/interfaces/round.v
theories/Classes/interfaces/ua_algebra.v
theories/Classes/interfaces/ua_congruence.v
theories/Classes/interfaces/ua_setalgebra.v
theories/Classes/isomorphisms/rings.v
theories/Classes/orders/archimedean.v
theories/Classes/orders/dec_fields.v
theories/Classes/orders/fields.v
theories/Classes/orders/integers.v
theories/Classes/orders/lattices.v
theories/Classes/orders/maps.v
theories/Classes/orders/nat_int.v
theories/Classes/orders/naturals.v
theories/Classes/orders/orders.v
theories/Classes/orders/rings.v
theories/Classes/orders/semirings.v
theories/Classes/orders/sum.v
theories/Classes/tactics/ring_pol.v
theories/Classes/tactics/ring_quote.v
theories/Classes/tactics/ring_tac.v
theories/Classes/tests/ring_tac.v
theories/Classes/theory/additional_operations.v
theories/Classes/theory/dec_fields.v
theories/Classes/theory/fields.v
theories/Classes/theory/int_abs.v
theories/Classes/theory/integers.v
theories/Classes/theory/lattices.v
theories/Classes/theory/nat_distance.v
theories/Classes/theory/naturals.v
theories/Classes/theory/premetric.v
theories/Classes/theory/rationals.v
theories/Classes/theory/ua_first_isomorphism.v
theories/Classes/theory/ua_homomorphism.v
theories/Classes/theory/ua_isomorphic.v
theories/Classes/theory/ua_prod_algebra.v
theories/Classes/theory/ua_quotient_algebra.v
theories/Classes/theory/ua_second_isomorphism.v
theories/Classes/theory/ua_subalgebra.v
theories/Classes/theory/ua_third_isomorphism.v
theories/Colimits/Sequential.v
theories/FunextAxiom.v
theories/HIT/surjective_factor.v
theories/Homotopy/CayleyDickson.v
theories/Homotopy/EMSpace.v
theories/Homotopy/HSpaceS1.v
theories/Metatheory/Core.v
theories/Metatheory/FunextVarieties.v
theories/Metatheory/IntervalImpliesFunext.v
theories/Metatheory/TruncImpliesFunext.v
theories/Metatheory/UnivalenceImpliesFunext.v
theories/Metatheory/UnivalenceVarieties.v
theories/PropResizing/ImpredicativeTruncation.v
theories/PropResizing/Nat.v
theories/Spaces/Card.v
theories/Spaces/Finite/FinSeq.v
theories/Spaces/List.v
theories/Spectra/Coinductive.v
theories/Tests.v
theories/UnivalenceAxiom.v
theories/Utf8.v
```
</details>
<details><summary>For comparison, files built while building HoTT.vo</summary>

```
theories/Algebra/AbGroups/AbelianGroup.v
theories/Algebra/AbGroups/Abelianization.v
theories/Algebra/AbGroups/AbPullback.v
theories/Algebra/AbGroups/AbPushout.v
theories/Algebra/AbGroups.v
theories/Algebra/AbGroups/Z.v
theories/Algebra/Aut.v
theories/Algebra/Congruence.v
theories/Algebra/Groups/FreeGroup/ListQuotient.v
theories/Algebra/Groups/FreeGroup.v
theories/Algebra/Groups/Group.v
theories/Algebra/Groups/GrpPullback.v
theories/Algebra/Groups/Image.v
theories/Algebra/Groups/Kernel.v
theories/Algebra/Groups/QuotientGroup.v
theories/Algebra/Groups/ShortExactSequence.v
theories/Algebra/Groups/Subgroup.v
theories/Algebra/Groups.v
theories/Algebra/ooAction.v
theories/Algebra/ooGroup.v
theories/Algebra/Rings/CRing.v
theories/Algebra/Rings/Ideal.v
theories/Algebra/Rings/QuotientRing.v
theories/Algebra/Rings.v
theories/Algebra/Rings/Z.v
theories/Basics/Contractible.v
theories/Basics/Datatypes.v
theories/Basics/Decidable.v
theories/Basics/Decimal.v
theories/Basics/Equivalences.v
theories/Basics/Hexadecimal.v
theories/Basics/Logic.v
theories/Basics/Nat.v
theories/Basics/Notations.v
theories/Basics/Numeral.v
theories/Basics/Overture.v
theories/Basics/PathGroupoids.v
theories/Basics/Tactics.v
theories/Basics/Trunc.v
theories/Basics/UniverseLevel.v
theories/Basics/Utf8.v
theories/Basics.v
theories/BoundedSearch.v
theories/Classes/interfaces/abstract_algebra.v
theories/Classes/interfaces/canonical_names.v
theories/Classes/theory/apartness.v
theories/Classes/theory/groups.v
theories/Classes/theory/rings.v
theories/Colimits/Colimit_Coequalizer.v
theories/Colimits/Colimit_Flattening.v
theories/Colimits/Colimit_Prod.v
theories/Colimits/Colimit_Pushout_Flattening.v
theories/Colimits/Colimit_Pushout.v
theories/Colimits/Colimit_Sigma.v
theories/Colimits/Colimit.v
theories/Colimits/MappingCylinder.v
theories/Colimits/Pushout.v
theories/Colimits/Quotient.v
theories/Colimits/SpanPushout.v
theories/Constant.v
theories/Cubical/DPathCube.v
theories/Cubical/DPathSquare.v
theories/Cubical/DPath.v
theories/Cubical/PathCube.v
theories/Cubical/PathSquare.v
theories/Cubical.v
theories/Diagrams/Cocone.v
theories/Diagrams/CommutativeSquares.v
theories/Diagrams/Cone.v
theories/Diagrams/ConstantDiagram.v
theories/Diagrams/DDiagram.v
theories/Diagrams/Diagram.v
theories/Diagrams/Graph.v
theories/Diagrams/ParallelPair.v
theories/Diagrams/Sequence.v
theories/Diagrams/Span.v
theories/DProp.v
theories/Equiv/BiInv.v
theories/EquivGroupoids.v
theories/Equiv/PathSplit.v
theories/Equiv/Relational.v
theories/ExcludedMiddle.v
theories/Extensions.v
theories/Factorization.v
theories/Functorish.v
theories/HFiber.v
theories/HIT/Coeq.v
theories/HIT/epi.v
theories/HIT/Flattening.v
theories/HIT/FreeIntQuotient.v
theories/HIT/Interval.v
theories/HIT/iso.v
theories/HIT/quotient.v
theories/HIT/SetCone.v
theories/HIT/unique_choice.v
theories/HIT/V.v
theories/Homotopy/BlakersMassey.v
theories/Homotopy/ClassifyingSpace.v
theories/Homotopy/ExactSequence.v
theories/Homotopy/Freudenthal.v
theories/Homotopy/HomotopyGroup.v
theories/Homotopy/HSpace.v
theories/Homotopy/Join.v
theories/Homotopy/Pi1S1.v
theories/Homotopy/Smash.v
theories/Homotopy/SuccessorStructure.v
theories/Homotopy/Suspension.v
theories/Homotopy/Wedge.v
theories/Homotopy/WhiteheadsPrinciple.v
theories/HoTT.v
theories/HProp.v
theories/HSet.v
theories/Idempotents.v
theories/Limits/Equalizer.v
theories/Limits/Limit.v
theories/Limits/Pullback.v
theories/Misc.v
theories/Modalities/Accessible.v
theories/Modalities/Closed.v
theories/Modalities/CoreflectiveSubuniverse.v
theories/Modalities/Descent.v
theories/Modalities/Fracture.v
theories/Modalities/Identity.v
theories/Modalities/Lex.v
theories/Modalities/Localization.v
theories/Modalities/Modality.v
theories/Modalities/Notnot.v
theories/Modalities/Nullification.v
theories/Modalities/Open.v
theories/Modalities/ReflectiveSubuniverse.v
theories/Modalities/Separated.v
theories/Modalities/Topological.v
theories/NullHomotopy.v
theories/ObjectClassifier.v
theories/PathAny.v
theories/Pointed/Core.v
theories/Pointed/Loops.v
theories/Pointed/pEquiv.v
theories/Pointed/pFiber.v
theories/Pointed/pHomotopy.v
theories/Pointed/pMap.v
theories/Pointed/pSusp.v
theories/Pointed/pTrunc.v
theories/Pointed.v
theories/Projective.v
theories/PropResizing/PropResizing.v
theories/Spaces/BAut/Bool/IncoherentIdempotent.v
theories/Spaces/BAut/Bool.v
theories/Spaces/BAut/Cantor.v
theories/Spaces/BAut/Rigid.v
theories/Spaces/BAut.v
theories/Spaces/Cantor.v
theories/Spaces/Circle.v
theories/Spaces/Finite/FinInduction.v
theories/Spaces/Finite/Finite.v
theories/Spaces/Finite/FinNat.v
theories/Spaces/Finite/Fin.v
theories/Spaces/Finite/Tactics.v
theories/Spaces/Finite.v
theories/Spaces/Int/Core.v
theories/Spaces/Int/Equiv.v
theories/Spaces/Int/LoopExp.v
theories/Spaces/Int/Spec.v
theories/Spaces/Int.v
theories/Spaces/Nat.v
theories/Spaces/No/Addition.v
theories/Spaces/No/Core.v
theories/Spaces/No/Negation.v
theories/Spaces/No.v
theories/Spaces/Pos/Core.v
theories/Spaces/Pos/Spec.v
theories/Spaces/Pos.v
theories/Spaces/Spheres.v
theories/Spaces/Torus/TorusEquivCircles.v
theories/Spaces/Torus/TorusHomotopy.v
theories/Spaces/Torus/Torus.v
theories/Spaces/TwoSphere.v
theories/Spaces/Universe.v
theories/Spectra/Spectrum.v
theories/Tactics/BinderApply.v
theories/Tactics/EquivalenceInduction.v
theories/Tactics/EvalIn.v
theories/Tactics/Nameless.v
theories/Tactics/RewriteModuloAssociativity.v
theories/Tactics.v
theories/Truncations/Connectedness.v
theories/Truncations/Core.v
theories/Truncations.v
theories/TruncType.v
theories/Types/Arrow.v
theories/Types/Bool.v
theories/Types/Empty.v
theories/Types/Equiv.v
theories/Types/Forall.v
theories/Types/Paths.v
theories/Types/Prod.v
theories/Types/Sigma.v
theories/Types/Sum.v
theories/Types/Unit.v
theories/Types/Universe.v
theories/Types.v
theories/Types/Wtype.v
theories/WildCat/Core.v
theories/WildCat/EmptyCat.v
theories/WildCat/EquivGpd.v
theories/WildCat/Equiv.v
theories/WildCat/Forall.v
theories/WildCat/FunctorCat.v
theories/WildCat/Induced.v
theories/WildCat/NatTrans.v
theories/WildCat/Opposite.v
theories/WildCat/Paths.v
theories/WildCat/PointedCat.v
theories/WildCat/Prod.v
theories/WildCat/Sigma.v
theories/WildCat/Square.v
theories/WildCat/Sum.v
theories/WildCat/TwoOneCat.v
theories/WildCat/Type.v
theories/WildCat/UnitCat.v
theories/WildCat.v
theories/WildCat/Yoneda.v
```
</details>

The parts that look suspicious to me are when some of one folder is needed for HoTT.vo, but some isn't.  E.g. Algebra, Classes, Colimits, HIT, Homotopy, etc.

**Comment by Alizter** [comment]
Yeah, I believe we left out everything from being included since we used to have notation clashes. But now I have fixed all the clashes so there is no reason to keep things from being included.

**Comment by mikeshulman** [comment]
I think certainly these should be included:
```
theories/Analysis/Locator.v
theories/Colimits/Sequential.v
theories/HIT/surjective_factor.v
theories/Homotopy/CayleyDickson.v
theories/Homotopy/EMSpace.v
theories/Homotopy/HSpaceS1.v
theories/Spaces/Card.v
theories/Spaces/Finite/FinSeq.v
theories/Spaces/List.v
```
I think it was originally intentional to leave out Categories and Classes.  I assume the parts of Classes that are built while building HoTT are imported by some other file, such as Algebra.  Do we want all of Algebra to be included in HoTT?  (I don't see why not.)  But even if so, we could still decline to *explicitly* include all of Classes.

I think Metatheory was intentionally excluded, as were the Axiom files, the two PropResizing files that are missing (since they are really more like metatheory too -- the "actual" truncation and nat we use in the library are defined as (H)ITs), and Spectra/Coinductive since it is unfinished/experimental. 

Actually, maybe Spectra/Coinductive should be deleted or moved to contrib?  And maybe PropResizing/ImpredicativeTruncation and PropResizing/Nat could be moved to Metatheory?  And we could put FunextAxiom and UnivalenceAxiom together into Axioms/Funext and Axioms/Univalence, which could then be extended further with say Axioms/PropResizing, etc.  Keeping all the HoTT-excluded files in specific subdirectories could help avoid this kind of problem recurring.

**Comment by mikeshulman** [comment]
Right.

**Comment by Alizter** [comment]
>Actually, maybe Spectra/Coinductive should be deleted or moved to contrib?
I think we intending on deleting it when working with spectra last year.

**Comment by jdchristensen** [comment]
Some things were fixed in #1548.  It turns out to be pretty tedious to try to improve things further, with little benefit, so I'll close this issue.  Feel free to reopen (or create a new issue) if you want to work on this more.  I created #1549 which records one specific task that seems worth doing.
